### PR TITLE
Spark 3

### PR DESCRIPTION
**Pull request by mhamilton723**

### CODE REVIEW

File: src/main/scala/com/microsoft/ml/spark/cognitive/AzureSearch.scala [hunk 1]
```diff
@@ -326,7 +326,7 @@ object AzureSearchWriter extends IndexParser with SLogging {
   }
 
   def write(df: DataFrame, options: Map[String, String] = Map()): Unit = {
-    prepareDF(df, options).foreachPartition(it => it.foreach(_ => ()))
```
Comment (eisber) [inline src/main/scala/com/microsoft/ml/spark/cognitive/AzureSearch.scala:329]: foreachPartition == foreach? surprising.

File: build.sbt [hunk 2]
```diff
@@ -269,19 +274,19 @@ genBuildInfo := {
...
+      |
+      |### Maven Coordinates
+      | `${organization.value}:${name.value}_${scalaMajorVersion}:${version.value}`
+      |
+      |### Maven Resolver
+      | `https://mmlspark.azureedge.net/maven`
+      |
+      |### Documentation Pages:
+      |[Scala Documentation](https://mmlspark.blob.core.windows.net/docs/${version.value}/scala/index.html)
+      |[Python Documentation](https://mmlspark.blob.core.windows.net/docs/${version.value}/pyspark/index.html)
```
Comment (eisber) [inline build.sbt:288]: will you also push to public maven?

File: build.sbt [hunk 3]
```diff
@@ -11,12 +11,15 @@ import scala.sys.process.Process
...
+val sparkVersion = "3.0.0"
 
-val sparkVersion = "2.4.5"
+//val scalaMajorVersion  = settingKey[String]("scalaMajorVersion")
+//scalaMajorVersion  := {scalaVersion.value.split(".".toCharArray).dropRight(0).mkString(".")}
+val scalaMajorVersion = 2.12
 
 val excludes = Seq(
-  ExclusionRule("org.apache.spark", "spark-tags_2.11"),
+  ExclusionRule("org.apache.spark", "spark-tags_2.12"),
```
Comment (eisber) [inline build.sbt:22]: isn't that scalaVersion dependent?

Comment (mhamilton723) [reply build.sbt:22]: Fixed, ty!

File: build.sbt (hunk 2 above)
Comment (mhamilton723) [reply build.sbt:288]: we did push to publicmaven but it broke, will need to go in there and figure out whats up

File: src/main/scala/com/microsoft/ml/spark/cognitive/AzureSearch.scala (hunk 1 above)
Comment (mhamilton723) [reply src/main/scala/com/microsoft/ml/spark/cognitive/AzureSearch.scala:329]: i think this is a ne3w func in spark3

Comment (eisber) [reply src/main/scala/com/microsoft/ml/spark/cognitive/AzureSearch.scala:329]: oh, so this only builds in spark3 now?

Comment (mhamilton723) [reply src/main/scala/com/microsoft/ml/spark/cognitive/AzureSearch.scala:329]: yes as you can see there is really no way to do both 2.4 and 3. Best best for 2.4 is to use latest release and we can make hotfixes on there too

**Comment by hoffrocket** [comment]
Hello,

Thanks so much for creating this library and working on this spark3 update. Are there any published artifacts for this branch that I could try now? Perhaps a SNAPSHOT version? I'm very eager to test it out. Thanks again!

**Comment by mhamilton723** [comment]
> Hello,
> 
> Thanks so much for creating this library and working on this spark3 update. Are there any published artifacts for this branch that I could try now? Perhaps a SNAPSHOT version? I'm very eager to test it out. Thanks again!

MMLSpark Build and Release Information
Maven Coordinates
com.microsoft.ml.spark:mmlspark_2.12:1.0.0-rc3-22-47c026d8-SNAPSHOT

Maven Resolver
https://mmlspark.azureedge.net/maven

Documentation Pages:
Scala Documentation 
Python Documentation

**Comment by hoffrocket** [comment]
> > Hello,
> > Thanks so much for creating this library and working on this spark3 update. Are there any published artifacts for this branch that I could try now? Perhaps a SNAPSHOT version? I'm very eager to test it out. Thanks again!
> 
> @hoffrocket here you go!
> 
> MMLSpark Build and Release Information
> Maven Coordinates
> com.microsoft.ml.spark:mmlspark_2.12:1.0.0-rc3-22-47c026d8-SNAPSHOT
> 
> Maven Resolver
> https://mmlspark.azureedge.net/maven
> 
> Documentation Pages:
> Scala Documentation
> Python Documentation

Thanks!

File: src/main/scala/com/microsoft/ml/spark/train/TrainRegressor.scala [hunk 4]
```diff
@@ -101,6 +101,7 @@ class TrainRegressor(override val uid: String) extends AutoTrainer[TrainedRegres
     val processedData   = featurizedModel.transform(convertedLabelDataset)
 
     processedData.cache()
+    processedData.show(100,false)
```
Comment (imatiach-msft) [inline src/main/scala/com/microsoft/ml/spark/train/TrainRegressor.scala:104]: remove?

File: src/test/scala/com/microsoft/ml/spark/nbtest/NotebookTests.scala [hunk 5]
```diff
@@ -19,27 +19,46 @@ class NotebookTests extends TestBase {
...
       installLibraries(clusterId)
-      tryWithRetries(Seq.fill(60*3)(1000).toArray){() =>
-        assert(isClusterActive(clusterId))}
+      tryWithRetries(Seq.fill(60 * 3)(1000).toArray) { () =>
+        assert(isClusterActive(clusterId))
+      }
       println(s"Creating folder $Folder")
       workspaceMkDir(Folder)
+      println(s"Submitting nonparallelizable job...")
+      val nonParFailutes = NonParallizableNotebooks.toIterator.map { nb =>
```
Comment (imatiach-msft) [inline src/test/scala/com/microsoft/ml/spark/nbtest/NotebookTests.scala:33]: can you put the non parallelizable VW notebooks at the end instead?

**Comment by mhamilton723** [comment]
For anyone looking to update to Spark 3 Please use the following coordinates. Thank you for your patience!


Maven Coordinates:
```com.microsoft.ml.spark:mmlspark_2.12:1.0.0-rc3-24-495af3e4-SNAPSHOT```
Maven Resolver:
```https://mmlspark.azureedge.net/maven```

**Pull request merged**